pyrk/th_component.py

```diff
@@ -83,33 +131,232 @@ def update_temp(self, timestep, temp):
         :param timestep: the timestep at which to query the temperature
         :type timestep: int
         :param temp: the new tempterature
-        :type float: float, units of kelvin
+        :type float: float, dimensionless
         """
         self.T[timestep] = temp
         self.prev_t_idx = timestep
         return self.T[timestep]
 
-    def dtemp(self):
-        if self.prev_t_idx == 0:
-            return 0.0*units.kelvin
-        else:
-            return (self.T[self.prev_t_idx] - self.T[self.prev_t_idx-1])
+    def dtemp(self, timestep, T0_timestep):
+        T0 = self.T[T0_timestep]
+        return self.T[timestep-1]-T0
 
-    def temp_reactivity(self):
-        return self.alpha_temp*self.dtemp()
+    def temp_reactivity(self, timestep, T0_timestep):
```

Comment:
Why add the timestep and the T0_tempstep? This function should get both of these values from the Timer object, if it needs them. No?